src/m365/pp/commands/solution/solution-remove.spec.ts

```diff
@@ -0,0 +1,235 @@
+const command: Command = require('./solution-remove');
+
+describe(commands.SOLUTION_REMOVE, () => {
+  let commandInfo: CommandInfo;
+  //#region Mocked Responses
+  const validEnvironment = '4be50206-9576-4237-8b17-38d8aadfaa36';
+  const validId = '00000001-0000-0000-0001-00000000009b';
+  const validName = 'Solution name';
+  const envUrl = "https://contoso-dev.api.crm4.dynamics.com";
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let promptOptions: any;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.delete,
+      powerPlatform.getDynamicsInstanceApiUrl,
+      Cli.prompt,
+      Cli.executeCommandWithOutput
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.SOLUTION_REMOVE), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if id is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        environment: validEnvironment,
+        id: 'Invalid GUID'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, id: validId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, name: validName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified solution owned by the currently signed-in user when confirm option not passed', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').callsFake(async () => envUrl);
```

Comment:
We are not executing any requests in this test, we can safely remove this `stub` since we don't need it in this test.